src/main/java/com/epam/aidial/core/controller/DeploymentPostController.java

```diff
@@ -66,10 +64,14 @@ public Future<?> handle(String deploymentId, String deploymentApi) {
             return context.respond(HttpStatus.UNSUPPORTED_MEDIA_TYPE, "Only application/json is supported");
         }
 
-        Deployment deployment = select(deploymentId, deploymentApi);
+        Deployment deployment = context.getConfig().selectDeployment(deploymentId);
+        if (!isValidDeploymentApi(deployment, deploymentApi)) {
+            deployment = null;
```

Comment:
Line 72. The deployment should be set to the context after the last check line 74

Reply:
Fixed